docs/docs/reference/02-yaml-syntax.md

```diff
@@ -0,0 +1,638 @@
+  component: <component_name>
+```
+
+Call a service with a string payload.
+```yaml
+service:
+  name: <service_name>
+  component: <component_name>
+  payload: "..."
+```
+
+The service payload can also be written as any standard YAML object. The application parser will automatically encode
+the object into a string format when making the service call. In this case, the component service is responsible
+for parsing the string back into a YAML object, dict or structure as necessary.
+
+```yaml
+service:
+  name: <service_name>
+  component: <component_name>
+  payload:
+    foo: "some content"
+    bar: [ x, y, z ]
+    baz:
+      a: 1
+      b: 2
+```
+
+#### Load or unload a hardware interface
+
+Load and initialize a hardware interface.
+```yaml
+load_hardware: <hardware_interface_name>
+```
+
+Unload and destroy a hardware interface.
+```yaml
+unload_hardware: <hardware_interface_name>
+```
+
+:::caution
+
+All hardware interfaces in the application are automatically loaded and initialized when the application starts.
+
+This behavior may change in the near future.
+
+:::
+
+#### Load or unload a controller
+
+```yaml
+load_controller:
+  interface: <hardware_interface_name>
+  controller: <controller_name>
+
+unload_controller:
+  interface: <hardware_interface_name>
+  controller: <controller_name>
+```
+
+Use a list to load or unload multiple controllers from a single predicate.
+```yaml
+load_controller:
+  - interface: <hardware_interface_name>
+    controller: controller_a
+  - interface: <hardware_interface_name>
+    controller: controller_b
+```
+
+#### Start or stop a controller
+
+Use the `switch_controllers` event to list the controllers to be started or stopped for a specific hardware interface.
+
+```yaml
+switch_controllers:
+  interface: <hardware_interface_name>
+  start: [ <controller_name>, <controller_name> ]
+  stop: [ "controller_three", "controller_four" ] 
+```
+
+:::note
+
+A controller must be loaded before it can be started, and must be stopped before it can be unloaded.
+
+:::
+
+### Special event predicates
+
+#### on_load
+
+The `on_load` predicate is provided by the state engine and set to true after the component
+has been loaded. Any events associated with the `on_load` predicate are handled once
+on instantiation of the node.
+
+```yaml
+component:
+  events:
+    on_load:
+      <some triggered event>: ...
+```
+
+#### on_unload
+
+The `on_unload` predicate is similar to the `on_load` predicate and is provided by the state engine.
+Any events associated with the `on_unload` predicate are handled once upon destruction of the component interface.
+
+```yaml
+component:
+  events:
+    on_unload:
+      <some triggered event>: ...
+```
+
+## Validating a YAML Application
+
+<!-- FIXME: link to the schema on GitHub once it is on main; relative paths will break if the doc is versioned -->
+The [YAML application schema](../../../schemas/applications/schema/application.schema.json) defines the structural rules
```

Comment:
Does it work at all when deployed on GitHub?

Reply:
Probably not. I want to release to main after this anyway, which will involve updating and tagging the schema version at which point it can be permalinked properly here. So it's a TODO but will be resolved soon.